src/VoteDelegate.sol

```diff
@@ -95,6 +104,11 @@ contract VoteDelegate {
         emit Free(msg.sender, wad);
     }
 
+    function reserveHatch() external {
+        require(block.number > hatchTrigger + HATCH_SIZE + HATCH_COOLDOWN, "VoteDelegate/cooldown-not-finished");
```

Comment:
consider asking for `urn` as input, and revert if the `urn` is not underwater (subject to a liquidation). This can mitigate a denial of service attack vector where malicious actor call `reserveHatch` just for fun, to fail pending `delegate` calls. 
Such an attack is not perfect as there is still a cooldown window the user can use, but front end users are lazy. and if their delegate call fails once, many will just give up.

two notes on such check:
1) might be enough to ask for an arbitrary `urn` without even checking the `ilk`. but if you don't expect many `MKR-A`, `MKR-B` etc, so it is best to verify the `ilk` of the `urn` is relevant.

2) it was pointed out that `reserveHatch` can also be used to prevent denial of service during a normal undelegate call, and not just during liquidations. if this is the case, then it shouldn't be a pre-condition to have an existing `urn` that is underwater.